shell/platform/android/io/flutter/embedding/android/FlutterView.java

```diff
@@ -421,6 +448,75 @@ protected void onSizeChanged(int width, int height, int oldWidth, int oldHeight)
     sendViewportMetricsToFlutter();
   }
 
+  @Override
+  protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
+    super.onLayout(changed, left, top, right, bottom);
+    setWindowManagerDisplayFeatures();
+  }
+
+  @Override
+  protected void onAttachedToWindow() {
+    super.onAttachedToWindow();
+    windowManager.registerLayoutChangeCallback(ContextCompat.getMainExecutor(getContext()),
+        windowManagerListener);
+    windowManager.registerDeviceStateChangeCallback(ContextCompat.getMainExecutor(getContext()),
+        windowManagerDeviceStateListener);
+  }
+
+  @Override
+  protected void onDetachedFromWindow() {
+    windowManager.unregisterLayoutChangeCallback(windowManagerListener);
+    windowManager.unregisterDeviceStateChangeCallback(windowManagerDeviceStateListener);
+    super.onDetachedFromWindow();
+  }
+
+  /**
+   * Refresh {@link androidx.window.WindowManager} and {@link android.view.DisplayCutout}
+   * display features. Fold, hinge and cutout areas are populated here.
+   */
+  private void setWindowManagerDisplayFeatures() {
+    List<DisplayFeature> displayFeatures = windowManager.getWindowLayoutInfo().getDisplayFeatures();
+    List<FlutterRenderer.DisplayFeature> result = new ArrayList<>();
+
+    // Data from androidx.window.WindowManager display features. Fold and hinge areas are
+    // populated here.
+    for (DisplayFeature displayFeature: displayFeatures){
+      Log.v(
+          TAG,
+          "WindowManager Display Feature reported with bounds = "
+              + displayFeature.getBounds().toString()
+              + " and type = "
+              + displayFeature.getType());
+      DisplayFeatureType type = DisplayFeatureType.UNKNOWN;
+      if (displayFeature.getType() == DisplayFeature.TYPE_FOLD) {
```

Comment:
Switch? If just these two, then I guess this is fine, but if there are more in the future, we should just switch on the `getType()`